nab/detectors/htmjava/src/main/java/nab/detectors/htmjava/HTMModel.java

```diff
@@ -60,15 +61,23 @@ public HTMModel(JsonNode modelParams) {
         // Get updated model parameters
         Parameters parameters = getModelParameters(modelParams);
 
+        HTMSensor<Object> sensor = (HTMSensor<Object>)Sensor.create(ObservableSensor::create,
+                SensorParams.create(SensorParams.Keys::obs, "ManualInput", supplier));
+
+        sensor.initEncoder(parameters);
+
+        int inputWidth = sensor.getEncoder().getN();
+        parameters.set(KEY.INPUT_DIMENSIONS, new int[]{inputWidth});
```

Comment:
Oh, I needed the number anyway for the potential radius... I think I might leave it as-is since it makes it clear that we're matching the potential radius to the input dimensions. And then a future change could maybe change the handling of the potential radius. (For example, maybe a "potentialRadius" of -1 should mean "the whole input space" and then all this pain and suffering can go away)